### PR TITLE
feat: support remote tables in the data loader

### PR DESCRIPTION
**Pull request by jackye1995**
`StreamingDataset`, `PermutationBuilder`, and `Permutation` now work against a `RemoteTable` (LanceDB Cloud and Enterprise), which unblocks benchmarking the loader against the enterprise cluster cache.

```python
db = lancedb.connect("db://my-db", api_key=..., host_override=...)
ds = StreamingDataset(db.open_table("training"), world_size=8, rank=r)
```

Rows are addressed by `_rowid` exactly as before — `PermutationReader::load_batch` already built the same `_rowid IN (...)` filter that `Table::take_row_ids` sends, so the loader's fetch was always the take path. It just was never allowed to run.

### The guard

`PermutationBuilder.__init__` rejected anything without `_inner`, so a `RemoteTable` raised `TypeError` before reaching the PyO3 layer — which already unwraps one via `_table._inner`.

### A bounded schema lookup

`PermutationReader::output_schema` reads the schema off a query plan, and building a plan on a remote table *executes* the query (`create_plan` → `execute_query`). With no limit that is `k = isize::MAX`, so asking a remote table for its output schema pulled the whole table over HTTP and threw it away — once per assigned split, on every epoch, since `StreamingDataset.__iter__` constructs a `Permutation` per split.

One row rather than zero, deliberately: lance gates its limit node on `self.limit.unwrap_or(0) > 0`, so `Some(0)` means *no limit*.

### Tables with an LSM write spec are refused

A permutation references rows by row id, and rows that have not been flushed to the base table do not have one yet. The loader could read around them, but they would then be missing from training with nothing said about it, so the build refuses such a table up front instead of half supporting it.

### Fallible identity construction

`PermutationReader::identity` resolved `inner_new` with `unwrap`. That was near total against a local dataset, but construction counts the base table — an HTTP round trip for a remote one — so a transient network or auth failure became a panic across the PyO3 boundary.

### Tests

End-to-end `permutation_builder` and `StreamingDataset` runs against a mock server, the former torch-free so it runs wherever the suite does, plus a test that a build succeeds without an LSM write spec and is refused once one is installed.

### CODE REVIEW

File: rust/lancedb/src/dataloader/permutation/builder.rs [hunk 1]
```diff
@@ -238,7 +238,26 @@ impl PermutationBuilder {
...
+                });
+            }
+            Ok(None) => {}
+            // Table types with no LSM write path cannot have a spec.
+            Err(Error::NotSupported { .. }) => {}
+            Err(err) => return Err(err),
+        }
+
+        // First pass, apply filter and load row ids.  `Shuffler` permutes positions, so
+        // every rank must scan the rows in the same order to build the same permutation.
```
[the remark this answers is not in the document]
Comment (jackye1995) [reply rust/lancedb/src/dataloader/permutation/builder.rs:260]: This one is known and accepted. `Shuffler` permutes positions, so the permutation is a function of (scan order, seed), and each rank builds its own — they agree only while the scan returns rows in the same order for the same table version.

The plan is to make the server return deterministic results for a given table version rather than canonicalize on the client. Remote support in the data loader is unreleased, so landing the client side ahead of that guarantee does not break anyone in the meantime.

File: rust/lancedb/src/dataloader/permutation/builder.rs [hunk 2]
```diff
@@ -238,7 +238,26 @@ impl PermutationBuilder {
 
     /// Builds the permutation table and stores it in the given database.
     pub async fn build(self) -> Result<Table> {
-        // First pass, apply filter and load row ids
+        // Rows that have not been flushed to the base table have no row id yet, so a
+        // permutation cannot address them.  Reading the base table alone would leave
+        // them out of training silently, so refuse the table instead.
+        match self.base_table.base_table().get_lsm_write_spec().await {
```
[the remark this answers is not in the document]
Comment (jackye1995) [reply rust/lancedb/src/dataloader/permutation/builder.rs:244]: Real gap, and now tracked internally.

The root cause sits below the loader: remote doesn't implement the Lazy consistency mode that native readers implicitly rely on. `read_consistency_interval` is a refresh throttle on native — `None` (the default) means Lazy, i.e. never refresh, so the handle is a stable snapshot. On remote, `compute_min_timestamp` turns the same value into a *minimum-freshness demand*: `None` sends no constraint, so the server serves latest. Same default, opposite meaning. `track_read_version` gives monotonic reads across query nodes, but nothing stops them moving forward onto a compacted version.

Worth noting the exposure isn't remote-only. `StreamingDataset.__setstate__` reopens the base table in every DataLoader worker, so native multi-worker already pins per worker at fork time rather than once. What remote changes is turning a one-shot pin into continuous re-resolution.

The mechanism to fix it already exists — `prepare_query_bodies` stamps `version` on every request from `current_version()`, so a handle that has checked out a concrete version is pinned end to end. The gap is that nothing in the loader calls `checkout`. Doing it properly means pinning on both backends and carrying the version across the worker fork, so it's a separate change rather than part of this PR. TODO added at the scan site.

File: rust/lancedb/src/dataloader/permutation/builder.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jackye1995) [reply rust/lancedb/src/dataloader/permutation/builder.rs:260]: Tracked internally for the server-side ordering guarantee, and referenced in a TODO at this line.

**Review by westonpace — APPROVED** [review]
